client/src/backoffice/InboxListPage.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { FC, useState, useEffect, ChangeEvent, useRef } from 'react';
+
+/* Libraries */
+import ReactModal from 'react-modal';
+import ReactPaginate from 'react-paginate';
+
+/* Helpers */
+import { Notification, getNotifications, getEvents, PoapEvent } from '../api';
+
+/* Components */
+import { Loading } from '../components/Loading';
+
+/* Assets */
+import plus from '../images/plus.svg';
+
+const PAGE_SIZE = 10;
+
+type PaginateAction = {
+  selected: number;
+};
+
+const InboxListPage: FC = () => {
+  const [page, setPage] = useState<number>(0);
+  const [total, setTotal] = useState<number>(0);
+  const [shouldResetPage, setShouldResetPage] = useState<boolean>(false);
+  const [notificationType, setNotificationType] = useState<string>('inbox');
+  const [recipientFilter, setRecipientFilter] = useState<string>('everyone');
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const [selectedEvent, setSelectedEvent] = useState<number | undefined>(undefined);
+  const [modalText, setModalText] = useState<string>('');
+  const [modalTitle, setModalTitle] = useState<string>('');
+  const [isFetchingNotifications, setIsFetchingNotifications] = useState<null | boolean>(null);
+  const [notifications, setNotifications] = useState<null | Notification[]>(null);
+  const [events, setEvents] = useState<PoapEvent[]>([]);
+
+  useEffect(() => {
+    fetchEvents();
+  }, []);
+
+  useEffect(() => {
+    page !== 0 ? setPage(0) : fetchNotifications();
+  }, [notificationType]);
+
+  useEffect(() => {
+    setShouldResetPage(true);
+  }, [recipientFilter, selectedEvent]);
+
+  useEffect(() => {
+    if (shouldResetPage) {
+      setShouldResetPage(false);
+
+      if (recipientFilter === 'event') {
+        if (selectedEvent === undefined) return;
+      } else {
+        setSelectedEvent(undefined);
+      }
+
+      page !== 0 ? setPage(0) : fetchNotifications();
+    }
+  }, [shouldResetPage]);
+
+  useEffect(() => {
+    fetchNotifications();
+  }, [page]);
+
+  const fetchEvents = async () => {
+    const events = await getEvents();
+    setEvents(events);
+  };
+
+  const fetchNotifications = () => {
+    setIsFetchingNotifications(true);
+
+    let event_id = undefined;
+    if (recipientFilter === 'event' && selectedEvent !== undefined) {
+      event_id = selectedEvent > -1 ? selectedEvent : undefined;
+    }
+
+    getNotifications(PAGE_SIZE, page * PAGE_SIZE, notificationType, event_id)
+      .then(response => {
+        if (!response) return;
+        setNotifications(response.notifications);
+        setTotal(response.total);
+      })
+      .catch(error => console.error(error))
+      .finally(() => setIsFetchingNotifications(false));
+  };
```

Comment:
Can we use `async/await` also here like you did in line [66](https://github.com/Xivis/poap/pull/37/files#diff-2308437293148e358de43d01f5ec572cR66) in order to have a homogeneous async logic avoiding mixing between async/await and promises.